plugins/pluginmanagers/lazy.nix

```diff
@@ -0,0 +1,216 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.lazy;
+  lazyPlugins = cfg.plugins;
+
+  processPlugin = plugin: let
+    mkEntryFromDrv = p:
+      if lib.isDerivation p
+      then {
+        name = "${lib.getName p}";
+        path = p;
+      }
+      else {
+        name = "${lib.getName p.pkg}";
+        path = p.pkg;
+      };
+    processDependencies =
+      if plugin ? dependencies && plugin.dependencies != null
+      then builtins.concatMap processPlugin plugin.dependencies
+      else [];
+  in
+    [(mkEntryFromDrv plugin)] ++ processDependencies;
+
+  processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
+  lazyPath = pkgs.linkFarm "lazy-plugins" processedPlugins;
+in {
+  options = {
+    plugins.lazy = {
+      enable = mkEnableOption "lazy.nvim";
+
+      plugins = with types; let
+        pluginType =
+          either
+          package
+          (submodule {
+            options = {
+              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
+
+              pkg = mkOption {
+                type = package;
+                description = "Vim plugin to install";
+              };
+
+              name = helpers.mkNullOrOption str "Name of the plugin to install";
+
+              dev = helpers.defaultNullOpts.mkBool false "When true, a local plugin directory will be used instead. See config.dev";
+
+              lazy = helpers.defaultNullOpts.mkBool true "When true, the plugin will only be loaded when needed. Lazy-loaded plugins are automatically loaded when their Lua modules are required, or when one of the lazy-loading handlers triggers";
+
+              enabled = helpers.defaultNullOpts.mkBool true "When false then this plugin will not be included in the spec";
+
+              cond =
+                helpers.mkNullOrOption
+                (oneOf [
```

Comment:
maybeRaw (either ....)

Reply:
either I am doing something wrong or this doesn't quite work out:
```
error: attempt to call something which is not a function but a set

       at /nix/store/i8d20zf065xdkjydxhclcr517ya0d09g-source/plugins/pluginmanagers/lazy.nix:59:17:

           58|               cond =
           59|                 helpers.mkNullOrOption
             |                 ^
           60|                 helpers.nixvimTypes.maybeRaw (either (listOf str) str)

```

Reply:
On second thought, reading the description again, I think this needs to be `maybeRaw bool` in any case...

Reply:
You are missing some parentheses